dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaManagerConnector.java

```diff
@@ -87,47 +91,84 @@ public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArg
     out.add(new ClouderaServicesTask());
     out.add(new ClouderaHostComponentsTask());
 
-    out.add(new ClouderaClusterCPUChartTask(1, HOURLY, REQUIRED));
-    out.add(new ClouderaClusterCPUChartTask(7, DAILY, OPTIONAL));
-    out.add(new ClouderaClusterCPUChartTask(30, DAILY, OPTIONAL));
-    out.add(new ClouderaClusterCPUChartTask(90, DAILY, OPTIONAL));
+    if (arguments.getStartDate() == null && arguments.getEndDate() == null) {
+      out.addAll(defaultTasks());
+    } else {
+      out.addAll(customDateRangeTasks(arguments.getStartDate(), arguments.getEndDate()));
+    }
+  }
 
-    out.add(new ClouderaHostRAMChartTask(1, HOURLY, REQUIRED));
-    out.add(new ClouderaHostRAMChartTask(7, DAILY, OPTIONAL));
-    out.add(new ClouderaHostRAMChartTask(30, DAILY, OPTIONAL));
-    out.add(new ClouderaHostRAMChartTask(90, DAILY, OPTIONAL));
+  private List<? extends Task<?>> defaultTasks() {
+    ZonedDateTime today = ZonedDateTime.of(LocalDateTime.now(), ZoneId.of("UTC"));
+    ZonedDateTime startDate = today.minusDays(90);
+    return ImmutableList.of(
+        (new ClouderaClusterCPUChartTask.Builder())
+            .setOutputFileName("cluster-cpu-90d.jsonl")
```

Comment:
We agreed that now with dates parameters we need only one file name, so all the logic here can be simplified.

Reply:
done